### ISSUE TITLE
This is what needs to be done

### ISSUE THREAD

**Issue by mickdekkers** [eve-apps/spaicheck#14]
# Things to add
- [ ] Create an admin page.
- [x] Add a whitelist editor to the admin page.
- [x] Add a 'Re-check' button next to the discard button on keys in the Import Errors section.
- [x] Handle duplicate key entry and other form errors -- display error to user using autoform's error system.
- [x] Detect a key's primary character automatically and allow it to be set manually in the Key Explorer.
- [x] Display primary character instead of key ID in key explorer.
- [x] Reject duplicate keys for the same account
- [ ] Allow users to add their email address and whether to receive email on the user settings modal.
- [ ] Provide undo option after discard/delete/modify operations.
- [x] On the Key Explorer, click the key to show details (like in the Import Errors section).
- [x] Merge the Validate All Keys section into the Key Explorer.
- [ ] Wrap Meteor.settings with methods for client use.
- [ ] Improve ongoworks:security integration for database writes.
- [x] Remove autopublish and manually define publishing/subscribing.
- [ ] Create getCharacterName Meteor method for use in Whitelist Editor and other places - check in DB and https://eveonline-third-party-documentation.readthedocs.org/en/latest/xmlapi/eve_charactername/ if not found
# Bugs to fix
- [x] When the Settings modal is closed before the "All settings saved" label begins fading out, it stays visible when the modal is re-opened.
- [ ] When the user is not on the whitelist (and not admin) and they manually navigate to restricted pages, or get removed from the whitelist and refresh the page, they do not get redirected and an error appears in the browser console: "Exception in setTimeout callback: Error: redirect needs to be done in sync".
# Things to test
- [x] What happens when a corporation key is added?
- [ ] What happens when a valid key becomes a corporation key?
- [ ] What happens when a large amount of keys are in the database?
# Code Quality fixes
- [x] Double check variable scope -- do we really need all those global variables?
- [x] Use single quotes or double quotes, but not both.
- [ ] Use "ID" or "Id", but not both (except where necessary).
# Things to add later (probably not in MVP)
- [ ] Add a Character Explorer to view various character information (put it where the Validate All Keys section is now).
- [ ] Add a section to modify app settings to the admin page (app settings TBD).
- [x] Add a bulk key import feature.
- [ ] Display import error recheck result using flash message

**Comment by detrohutt** [comment]
I think instead of "Add confirmation dialogs to discard/delete operations" we should give an undo option, maybe using flash-messages. What do you think?
http://ux.stackexchange.com/questions/71960/deletion-confirm-or-undo-which-is-the-better-option-and-why

**Comment by mickdekkers** [comment]
Yes, definitely. Good idea, undo is the better option. It would also probably be more straightforward to implement undo than confirm, since with the latter you'd have to stop the flow and wait for a user response to do anything.
